user_guide_src/source/concepts/factories.rst

```diff
@@ -0,0 +1,117 @@
+#########
+Factories
+#########
+
+.. contents::
+    :local:
+    :depth: 2
+
+Introduction
+============
+
+Like ``Services``, ``Factories`` are an extension of autoloading that helps keep your code
+concise yet optimal, without having to pass around object instances between classes. At its
+simplest, ``Factories`` provides a common way to create a class instance and access it from
+anywhere. This is a great way to reuse object states and reduce memory load from keeping
+multiple instances loaded across your app.
+
+Anything can be loaded from a Factory, but the best examples are those classes that are used
+to work on or transmit common data. The framework itself makes heavy use of the ``Config``
+Factory to make sure the correct configuration is in use at any time. 
+
+Take a look at ``Models`` as an example. You can access the Factory specific to ``Models``
+by using the magic static method of the Factories class, ``Factories::models()``. Because
+of the common path structure for namespaces and folders, ``Factories`` knows to look for
+files and classes within ``Models``, so you can request a model by its shorthand base name::
+
+	use CodeIgniter\Config\Factories;
+
+	$users = Factories::models('UserModel');
+
+Or you could also request a specific class::
+
+	$widgets = Factories::models('Some\Namespace\Models\WidgetModel');
+
+Next time you ask for the same class anywhere in your code, ``Factories`` will be sure
+you get back the instance as before::
+
+	class SomeOtherClass
+	{
+		$widgets = Factories::models('WidgetModel');
+
+
+Factories Behavior
+==================
+
+The default behavior of ``Factories`` might not always work for ever component. For example,
+say your component name and its path do not align, or you need to be sure that only a certain
+type of class instances are returned, or your component classes require additional parameters
+for their constructor. Learn how to use ``Factories`` and its configuration to meet your needs.
```

Comment:
```suggestion
The default behavior of ``Factories`` might not always work for every component. For example,
say your component name and its path do not align, or you need to be sure that only a certain
type of class instance is returned, or your component classes require additional parameters
for their constructor. Learn how to use ``Factories`` and its configuration to meet your needs.
```


> or your component classes require additional parameters for their constructor.

Isn't the additional arguments for constructor handled by the splat operator (`...`)? I mean after the two `array_shift`, what is left of the `$arguments` is unpacked to the constructor.

Reply:
That's correct. The header section alludes to the Parameters section that covers this. I reworded it a bit so hopefully it is more clear.